tools/src/test/java/org/apache/kafka/tools/consumer/group/ResetConsumerGroupOffsetTest.java

```diff
@@ -62,506 +86,756 @@
+        return topics.stream()
+                .collect(toMap(Function.identity(),
+                        topic -> singletonMap(new TopicPartition(topic, 0),
+                                expectedOffset)));
     }
 
-    private Map<String, Map<TopicPartition, OffsetAndMetadata>> resetOffsets(ConsumerGroupCommand.ConsumerGroupService consumerGroupService) {
+    private Map<String, Map<TopicPartition, OffsetAndMetadata>> resetOffsets(
+            ConsumerGroupCommand.ConsumerGroupService consumerGroupService) {
         return consumerGroupService.resetOffsets();
     }
 
-    Map<TopicPartition, Long> toOffsetMap(Map<TopicPartition, OffsetAndMetadata> map) {
-        return map.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().offset()));
+    private Map<TopicPartition, Long> partitionToOffsets(String topic,
+                                                         Map<TopicPartition, OffsetAndMetadata> partitionInfo) {
+        return partitionInfo.entrySet()
+                .stream()
+                .filter(entry -> Objects.equals(entry.getKey().topic(), topic))
+                .collect(toMap(Map.Entry::getKey, e -> e.getValue().offset()));
+    }
+
+    private static List<String> generateIds(String name) {
+        return IntStream.rangeClosed(1, 2)
+                .mapToObj(id -> name + id)
+                .collect(Collectors.toList());
+    }
+
+    private void produceConsumeAndShutdown(ClusterInstance cluster,
+                                           String topic,
+                                           String group,
+                                           int numConsumers,
+                                           GroupProtocol groupProtocol) throws Exception {
+        produceMessages(cluster, topic, 100);
+        try (AutoCloseable consumerGroupCloseable =
+                     consumerGroupClosable(cluster, numConsumers, topic, group, groupProtocol)) {
+            awaitConsumerProgress(cluster, topic, group, 100);
+        }
+    }
+
+    private AutoCloseable consumerGroupClosable(ClusterInstance cluster,
+                                                int numConsumers,
+                                                String topic,
+                                                String group,
+                                                GroupProtocol groupProtocol) {
+        Map<String, Object> configs = composeConsumerConfigs(cluster, group, groupProtocol);
+        return ConsumerGroupCommandTestUtils.buildConsumers(
+                numConsumers,
+                false,
+                topic,
+                () -> new KafkaConsumer<String, String>(configs));
+    }
+
+    private Map<String, Object> composeConsumerConfigs(ClusterInstance cluster,
+                                                       String group,
+                                                       GroupProtocol groupProtocol) {
+        HashMap<String, Object> configs = new HashMap<>();
```

Comment:
This configs could make the test slower. In those test cases they expect to wait "committed consumer offsets", and hence we have to wait (balance time: 3s + internal commit time: 5s) at least ...

Maybe we can reduce the value of both `group.initial.rebalance.delay.ms` and `auto.commit.interval.ms` to "1000"? WDYT?